src/multipart.rs

```diff
@@ -307,17 +307,59 @@ impl Stream for Multipart {
                 }
             };
 
-            if &boundary_bytes[..]
-                == format!("{}{}{}", constants::BOUNDARY_EXT, boundary, constants::BOUNDARY_EXT).as_bytes()
-            {
+            if &boundary_bytes[..] == format!("{}{}", constants::BOUNDARY_EXT, boundary).as_bytes() {
+                state.stage = StreamingStage::DeterminingBoundaryType;
+            } else {
+                return Poll::Ready(Some(Err(crate::Error::IncompleteStream)));
+            }
+        }
+
+        if state.stage == StreamingStage::DeterminingBoundaryType {
+            let ext_len = constants::BOUNDARY_EXT.len();
+            let next_bytes = match stream_buffer.peek_exact(ext_len) {
+                Some(bytes) => bytes,
+                None => {
+                    return if stream_buffer.eof {
+                        Poll::Ready(Some(Err(crate::Error::IncompleteStream)))
+                    } else {
+                        Poll::Pending
+                    };
+                }
+            };
+
+            if next_bytes == constants::BOUNDARY_EXT.as_bytes() {
                 state.stage = StreamingStage::Eof;
                 return Poll::Ready(None);
+            } else {
+                state.stage = StreamingStage::ReadingTransportPadding;
             }
+        }
 
-            if &boundary_bytes[..] != format!("{}{}{}", constants::BOUNDARY_EXT, boundary, constants::CRLF).as_bytes() {
-                return Poll::Ready(Some(Err(crate::Error::IncompleteStream)));
-            } else {
+        if state.stage == StreamingStage::ReadingTransportPadding {
+            if !stream_buffer.read_transport_padding() {
```

Comment:
The naming of `read_transport_padding` confused me. This block seems to say: if you don't read transport padding and we're not at EOF, go into pending. That is, require transport padding.

But really, we just want to keep skipping padding until there's something that's not LWSP. So really, `!read_transport_padding` is saying "until there's non-LWSP". Can we name it something like that?

Reply:
Naming this function was tough. Would something like `eat_transport_padding` or `advance_past_transport_padding` be better?